src/controllers/ride/deleteRide.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Response } from "express";
+import { rideRepository } from "../../repositories/rideRepository";
+import { Ride } from "../../entity/Ride";
+import { userRepository } from "../../repositories/userRepository";
+import { User } from "../../entity/User";
+import { z } from "zod";
+import { validate } from "../../helpers/zodValidateRequest";
+
+const dataSchema = z.object({
+  params: z.object({
+      id: z
+        .string({
+          invalid_type_error: "id not a string",
+          required_error: "id is a required parameter",
+        })
+        .min(0, {
+          message: "id must be a non-empty string",
+        })
+        .uuid({ message: "id must be a valid uuid" }),
+    })
+  })
+
+export const deleteRideValidator = validate(dataSchema)
+
+export const deleteRide = async (req: Request, res: Response) => {
+  const rideId = req.params.id;
+  const userId = req.token._id;
+
+  const userObj: User = await userRepository
+    .createQueryBuilder("user")
+    .where("user.id = :id", { id: userId })
+    .getOne()
+
+  let rideObj: Ride | null = null;
+
+  try {
+    rideObj = await rideRepository
+      .createQueryBuilder('ride')
+      .where('ride.id = :id', { id: rideId })
+      .getOne();
+
+    if (!rideObj) {
+      return res.status(404).json({ message: "Ride not found in DB" });
+    }
+
+    if (userObj == rideObj.originalPoster) {
```

Comment:
can't you just compare userId directly and save one extra request to the db (the one where you created `userObj`)?